packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -1542,6 +1545,32 @@ export class Project extends ProjectBase {
       throw new Error('Both source and artifacts must be set to CodePipeline');
     }
   }
+
+  private isLambdaBuildImage(buildImage: IBuildImage): boolean {
+    return buildImage instanceof LinuxLambdaBuildImage || buildImage instanceof LinuxArmLambdaBuildImage;
+  }
+
+  /**
+   * Validates a Lambda build image given the project properties.
+   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/lambda.html#lambda.limitations
+   */
+  private validateLambdaBuildImage(buildImage: IBuildImage, props: ProjectProps): string[] {
+    if (!this.isLambdaBuildImage(buildImage)) return [];
+    const ret = [];
+    if (props.timeout) {
```

Comment:
```suggestion
    if (buildImage.imagePullPrincipalType) {
      ret.push('Cannot specify imagePullPrincipalType for Lambda compute images. Lambda compute doesn\'t support custom images.');
    }
    if (buildImage.repository) {
      ret.push('Cannot specify repository for Lambda compute images. Lambda compute doesn\'t support custom images.');
    }
    if (buildImage.secretsManagerCredentials) {
      ret.push('Cannot specify secretsManagerCredentials for Lambda compute images. Lambda compute doesn\'t support custom images.');
    }
    if (props.timeout) {
```
We should just validate `imagePullPrincipalType`, `repository`, and `secretsManagerCredentials` for the Lambda compute image (will also need related unit tests).

Reply:
Previously, setting `imagePullPrincipalType` to `CODEBUILD` seemed to cause an error (really sorry if I misunderstood). However, I just rechecked the behavior and found that deploying with `imagePullPrincipalType` set to `CODEBUILD` succeeds.
Therefore, I've changed the default value of `imagePullPrincipalType` in the `LinuxLambdaBuildImage` and `LinuxArmLambdaBuildImage` classes to `CODEBUILD`.

For `repository` and `secretsManagerCredentials`, these properties are not specified in the Lambda Build Image and are read-only, i.e., they cannot be changed after initialization in the Lambda Build Image. So, I don't think I can write unit tests to check for validation errors.
Do you mean to add own class that implements IBuildImage (like the lambda build image classes) and test it?